File: backend/app/shared/permissions.py
from enum import StrEnum


class Role(StrEnum):
    OWNER = "owner"
    ADMIN = "admin"
    MEMBER = "member"


ROLE_RANK: dict[Role, int] = {
    Role.MEMBER: 0,
    Role.ADMIN: 1,
    Role.OWNER: 2,
}


def has_minimum_role(actual: Role, required: Role) -> bool:
    return ROLE_RANK[actual] >= ROLE_RANK[required]

File: frontend/src/lib/permissions.ts
import type { Role } from "@/types/auth";

/** Mirrors the backend's `require_role(Role.ADMIN)` check (OWNER/ADMIN can write, MEMBER is read-only). */
export function canWrite(role: Role | undefined): boolean {
  return role === "owner" || role === "admin";
}
